### DISCUSSION TITLE
Development of new control software for SLA/DLP printers

### DISCUSSION THREAD

**Discussion by arslan437** [sn4k3/UVtools#639] [General]
Hi, 

So I have an idea to make an DLP/SLA printer without any raspberry pi or SBC. 

### With the following features

- Easy to use 
- Load the stl and just press the print(All the slicing and other stuff will be done by the software it self)
- Initially hardware is to controlled by the marlin. Later to develop a new or strip the marlin to custom firmware. 
- Software to run on windows 
- it will detect the connected display with the computer 
- When user press the start button it will slice the file and home the machine than display the image/pattent on the screen and move z axis and so on.  

### Currently struggling at 

- At this time my plan is to use the python to build the gui and other stuff. Because it will be much faster to build. 
- I don't want to build the slicer. looking for a Command line slicer. Maybe pursa will work. But don't know if we can change the build plate size for sla printers or not. 
- Than can I open the sliced file. I thing your software will help alot in this case.   

I am looking for your idea and if any one have any suggestion on it. Do you guys thing is it going to be a good project or not. If I can do this then the SLA/DLP printers will much easier to build as compared to FDM printers.

**Comment by sn4k3** [comment]
As a personal hobby project I think it's great, however if for comercial product or users itself I think it's not good.

> Load the stl and just press the print(All the slicing and other stuff will be done by the software it self)

It's fine to have that feature, PrusaSlicer can do it via cmd, but if you take the freedom from users from place it own supports people will not like it, beside that there are important stuff to control and tune. Also don't forget auto supports will fail somewhere.
Again it's fine to have that feature IF you also allow people to send it own sliced file.

> Software to run on windows

This is a bad chooise. Windows is paied, slower than alternatives and risky. It's fine to support windows, but if you will do a dedicated host, do it with a stable LTS Linux Server.

> it will detect the connected display with the computer

This of course will only happen for LCDs with HDMI interfaces.

Having dedicated host will ramp price and people are after cheap stuff, you may think everyone has a spare and old laptop, most will not and still most of old laptops can't do even 4K. So you will require a capable host which cost some good amount of cash, maybe more than a printer in these days.

You still need an motion board, so your PC must comunicate with something that handles the printer hardware and then more cost is involved.

This is not to demotivate you, but rather think in this more deeply.
